lib/semantics/resolve-labels.cc

```diff
@@ -898,13 +898,11 @@ void CheckLabelDoConstraints(const SourceStmtList &dos,
               TargetStatementEnum::CompatibleDo)) {
         errorHandler.Say(doTarget.parserCharBlock,
             parser::MessageFormattedText{
-                "Only an END DO or CONTINUE must be used to terminate a labeled DO loop"_err_en_US,
-                SayLabel(label)});
+                "END DO or CONTINUE must be used to terminate a labeled DO loop"_err_en_US});
```

Comment:
The irony of this message is that code like
```
    DO 10 i = 1,10
10 IF (a.eq.b) THEN
```
can be repaired to compile by disobeying the directive.
```
    DO 10 i = 1,10
10   arr(i) = d
    IF (a.eq.b) THEN
```

Reply:
Good point.  There are (at least) three compilation cases depending on which options are in effect.  First, the code (as illustrated by your example) might simply be illegal.  For the other two cases, the code might be something that we want to support but is illegal in the Fortran 2018 standard.  If the -Mstandard option is in effect, we would want to issue a message about END DO and CONTINUE that's a warning.  If the -Werror option is also in effect, that message should be an error.

Doing this differentiation is beyond the scope of what I intended when I created issue 392.  I'll create another issue to do this differentiation.